service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -937,9 +1004,13 @@ public ResponseEntity<ApiCloneReferencedGitRepoResourceResult> cloneGitRepoRefer
   @Override
   public ResponseEntity<ApiTerraWorkspaceResource> createTerraWorkspaceReference(
       UUID workspaceUuid, @Valid ApiCreateTerraWorkspaceReferenceRequestBody body) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    workspaceService.validateWorkspaceAndAction(
+        userRequest, workspaceUuid, SamWorkspaceAction.CREATE_REFERENCE);
     UUID referencedWorkspaceId = body.getReferencedWorkspace().getReferencedWorkspaceId();
 
-    // Will throw if workspace does not exist or workspace id is null.
+    // Will throw if the referenced workspace does not exist or workspace id is null.
+    // Note that the user does not need read access in the destination workspace to reference it.
```

Comment:
@melissachang just to confirm expected behavior: WSM should throw a 404 if the referenced workspace doesn't exist, but it's fine if the caller isn't a member of the referenced workspace (i.e. don't throw a 403 in that case)?

Reply:
Yes, that's what Yu and Dan settled on. Your comment changes look good, thanks for clarifying.

Reply:
I thought it would look silly if we did not test existence within our own database! Not testing access is consistent with the other referenced resources.

Reply:
sgtm, I just wanted to clarify what was expected after I tripped over a few tests!